### ISSUE TITLE
Plugin fails with "Maximum call stack size exceeded"

### ISSUE THREAD

**Issue by shogerr** [giddyinc/mongoose-to-swagger#12]
The mongoose-to-swagger plugin fails when using the minimal example provided by the author with the error `RangeError: Maximum call stack size exceeded`.

**System**
Window 10
node v13.5.0
yarn 1.21.1

**Test File**
```
// generate_swager.js
const mongoose = require('mongoose');
const m2s = require('mongoose-to-swagger');
const Cat = mongoose.model('Cat', { name: String });
const swaggerSchema = m2s(Cat);
console.log(swaggerSchema);
```

**Steps to Recreate**
```
PS C:\> yarn init mongoose-to-swagger-test -y
PS C:\> cd .\mongoose-to-swagger-test
PS C:\mongoose-to-swagger-test> yarn add mongoose mongoose-to-swagger
PS C:\mongoose-to-swagger-test> node generate_swagger.js

PS C:\mongoose-to-swagger-test> node .\generate_swagger.js
C:\mongoose-to-swagger-test\node_modules\mongoose-to-swagger\dist\index.js:113
const getFieldsFromMongooseSchema = (schema, options) => {
                                    ^

RangeError: Maximum call stack size exceeded
    at getFieldsFromMongooseSchema (C:\mongoose-to-swagger-test\node_modules\mongoose-to-swagger\dist\index.js:113:37)
    at mapSchemaTypeToFieldSchema (C:\mongoose-to-swagger-test\node_modules\mongoose-to-swagger\dist\index.js:98:22)
    at getFieldsFromMongooseSchema (C:\mongoose-to-swagger-test\node_modules\mongoose-to-swagger\dist\index.js:121:23)
    at mapSchemaTypeToFieldSchema (C:\mongoose-to-swagger-test\node_modules\mongoose-to-swagger\dist\index.js:98:22)
    at getFieldsFromMongooseSchema (C:\mongoose-to-swagger-test\node_modules\mongoose-to-swagger\dist\index.js:121:23)
    at mapSchemaTypeToFieldSchema (C:\mongoose-to-swagger-test\node_modules\mongoose-to-swagger\dist\index.js:98:22)
    at getFieldsFromMongooseSchema (C:\mongoose-to-swagger-test\node_modules\mongoose-to-swagger\dist\index.js:121:23)
    at mapSchemaTypeToFieldSchema (C:\mongoose-to-swagger-test\node_modules\mongoose-to-swagger\dist\index.js:98:22)
    at getFieldsFromMongooseSchema (C:\mongoose-to-swagger-test\node_modules\mongoose-to-swagger\dist\index.js:121:23)
    at mapSchemaTypeToFieldSchema (C:\mongoose-to-swagger-test\node_modules\mongoose-to-swagger\dist\index.js:98:22)
    at getFieldsFromMongooseSchema (C:\mongoose-to-swagger-test\node_modules\mongoose-to-swagger\dist\index.js:121:23)
    at mapSchemaTypeToFieldSchema (C:\mongoose-to-swagger-test\node_modules\mongoose-to-swagger\dist\index.js:98:22)
    at getFieldsFromMongooseSchema (C:\mongoose-to-swagger-test\node_modules\mongoose-to-swagger\dist\index.js:121:23)
    at mapSchemaTypeToFieldSchema (C:\mongoose-to-swagger-test\node_modules\mongoose-to-swagger\dist\index.js:98:22)
    at getFieldsFromMongooseSchema (C:\mongoose-to-swagger-test\node_modules\mongoose-to-swagger\dist\index.js:121:23)
    at mapSchemaTypeToFieldSchema (C:\mongoose-to-swagger-test\node_modules\mongoose-to-swagger\dist\index.js:98:22)
```

**Comment by Zwimber** [comment]
Same problem, originated after upgrading Mongoose

**Comment by blugavere** [comment]
fixed in mongoose-to-swagger@1.0.1 - pls verify

**Comment by shogerr** [comment]
Great success, everything tests and runs, and the ouput is as expected:
```
node .\generate_swagger.js
{
  title: 'Cat',
  properties: { name: { type: 'string' }, _id: { type: 'string' } }
}
```
